### PR TITLE
Publish button

### PR DESCRIPTION
**Pull request by JoshBudzynski**
Issue #18 

Creates a Button that has an action connected to it. As the instructions detail. 

Needs to add the updating of a GitHub, looking for some guidance on how to do this and what I am updating.

### CODE REVIEW

File: .github/Button.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    class PublishButton extends HTMLElement {
+      constructor() {
+        super();
+        const template = document.getElementById('publish-button-template').content;
+        this.attachShadow({ mode: 'open' }).appendChild(template.cloneNode(true));
+      }
+
+      connectedCallback() {
+        const button = this.shadowRoot.getElementById('publish-button');
+        button.addEventListener('click', async () => {
```
Comment (tluke900) [inline .github/Button.html:64]: No implementation for Git Pushes, listener is present for such actions however in the future.

File: .github/Button.html [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    body {
+      font-family: Arial, sans-serif;
+      padding: 20px;
+    }
+
+    h1 {
+      color: #333;
+    }
+
+    /* Style the button */
```
Comment (tluke900) [inline .github/Button.html:18]: Correctly matches specifications of the button

File: .github/Button.html [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        const template = document.getElementById('publish-button-template').content;
+        this.attachShadow({ mode: 'open' }).appendChild(template.cloneNode(true));
+      }
+
+      connectedCallback() {
+        const button = this.shadowRoot.getElementById('publish-button');
+        button.addEventListener('click', async () => {
+          // For now, simulate the publish action
+          alert('Publish action triggered');
+        // I dont know the idea behind this quite yet looking for tips
```
Comment (tluke900) [inline .github/Button.html:67]: Once you begin implementing the Git Push, create a arbitrary directory to push too for now.

**Review by tluke900 — COMMENTED** [review]
Only significant missing piece is the git push.

File: .github/Button.html
Comment (kate-holdener) [inline .github/Button.html:1]: Please move this button out of the .github directory

**Review by jacksayshi — CHANGES_REQUESTED** [review]
Thank you for working on this issue.
Here are few things to get this merged Please move HTML out of .github and for publishing to github "data" branch we can use a nodejs server with library like simplegit.

File: Data_Directory/.DS_Store
Comment (kate-holdener) [inline Data_Directory/.DS_Store:1]: Please remove this file and the Data_Directory

File: .DS_Store
Comment (kate-holdener) [inline .DS_Store:1]: Please remove this file

File: BoneServer.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  try {
+    const directoriesToCommit = [
+      path.join(__dirname, 'Data_Directory', 'Annotation_Directory'), // Correct path to Annotation_Directory
+      path.join(__dirname, 'Data_Directory', 'Bone_Directory'),       // Correct path to Bone_Directory
+      path.join(__dirname, 'Data_Directory', 'Image_Directory')       // Correct path to Image_Directory
+    ];
+
+    await git.add(directoriesToCommit);
+    await git.commit('Automated publish commit');
+    await git.push('origin', 'Publish_Button');
```
Comment (kate-holdener) [inline BoneServer.js:26]: Your push should be going here: https://github.com/oss-slu/DigitalBonesBox/tree/data/databones

**Review by jacksayshi — CHANGES_REQUESTED** [review]
Thank you for working on the changes.
As required it is publishing changes to git. But, we need that changes to be pushed not to the branch you have created and worked on. It should be Origin "data" branch. 
If you could make that change you also wouldnt need Data_Directory. 
Please make these changes to get this merged

**Pull request closed without merging**